src/utils.py

```diff
@@ -31,10 +33,25 @@ def get_mac_address():
 
 # Get IP address
 def get_ip_address():
-    socket_ = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-    socket_.connect((DNS_SERVER, DNS_PORT))
-    ip_address = socket_.getsockname()[0]
-    socket_.close()
+    """
+    Get the external IP address from the Balena supervisor API.
+    """
+    ip_address = None
+    try:
+        balena_api_url = f'{BALENA_SUPERVISOR_ADDRESS}/v1/device'\
```

Comment:
I thought this was the balena cloud API, and thus thought it was internet-dependent. Add to the docstring that the Balena supervisor is a local API?